File: Self Dividing Numbers.py
#

from os import *
from sys import *
from collections import *
from math import *

def findAllSelfDividingNumbers(lower, upper):

    # Write your code here
    # This function returns a list of all the self-dividing numbers in the range [left, right]
    result=[]
    for i in range(lower,upper+1):
        val=i
        while val:
            rem=val%10
            if rem!=0:
                if i%(rem)==0:
                    val//=10
                else:
                    break
            else:
                break
        if val==0:
            result.append(i)
    return result

File: Self Dividing Numbers.cpp
#

#include <bits/stdc++.h> 
vector < int > findAllSelfDividingNumbers(int lower, int upper) {
    // Write your code here
    vector <int> result;
    for (int i=lower;i<=upper;i++){
        int k=i;
        int possible=1;
        while (k){
            int rem=k%10;
            k=k/10;
            if (rem==0){
                possible=0;
                break;
            }
            if(i%rem!=0){
                possible=0;
                break;
            }
            
        }
        if (possible==1){
            result.push_back(i);
        }
    }
    return result;
}
